### DISCUSSION TITLE
Add divider for items in toolbar.

### DISCUSSION THREAD

**Discussion by XMB-7** [chrisgurney/obsidian-note-toolbar#99] [Ideas]
Like what `Editing Toolbar` does.
![image](https://github.com/chrisgurney/obsidian-note-toolbar/assets/65625632/ce72a37f-5f60-4f09-afad-ce74504d251f)

Besides adding a divider, could you also please consider allowing users to insert other markers like `\n` to place certain items on new lines? Thanks.
I currently use such a workaround to do this.
![image](https://github.com/chrisgurney/obsidian-note-toolbar/assets/65625632/ee7192d0-1f02-4117-9391-5d102dd59f23)

**Comment by chrisgurney** [comment]
@XMB-7 thanks for the ideas!

I have _separators_ on my list, as I'll be working on embedding toolbars (as "button groups") next, and want a way to visually separate them. I have yet to think through styling and defaults for separators, but the goal is to make it be configurable through [Style Settings](https://github.com/chrisgurney/obsidian-note-toolbar/wiki/Style-Settings-plugin-support).

As for _new lines_, would you consider instead using (1) a [Menu](https://github.com/chrisgurney/obsidian-note-toolbar/releases/tag/1.9.0-beta-1) or (2) perhaps a second toolbar that appears under the first on clicking an item? I've had requests for number 2 and will be exploring the possibility in the hopefully near future.

**Reply by XMB-7** [reply]
Maybe a second toolbar better suits my use case. Looking forward to this new feature. Thanks.

**Reply by chrisgurney** [reply]
Another thought I had: Maybe there should be a setting as to how toolbars wrap, and if you had a button group that doesn't fit on the line, it would move down to the next line. In that case it would work out for your example, without the additional click/tap.

**Comment by chrisgurney** [comment]
@XMB-7 Separators are now available in the [beta release](https://github.com/chrisgurney/obsidian-note-toolbar/releases/tag/1.10.0-beta-11), if you'd like to try it via BRAT. It's recommended that you try this in another vault or backup your data.

**Reply by XMB-7** [reply]
Thanks for supporting this feature. But I just found in the sandbox vault that after adding a separator clicking on 'adding toolbar item' button would also adding a separator unexpectedly. Besides, it seems that there is no way to convert a separator item into a normal item. 
![recording](https://github.com/user-attachments/assets/a679ff6f-3bb5-460a-8467-ddbb10331bf1)

**Reply by chrisgurney** [reply]
@XMB-7 Thanks for checking it out! 

~~What theme are you using?~~ You said you're using a sandbox vault right? And nothing else is installed perhaps aside from BRAT?

~~If I can't reproduce the issue,~~ it might help if you could run the `Show debug info` command and copy/paste it here.

(It won't be possible to convert to/from separator items.)

**Reply by XMB-7** [reply]
Yes, I tested it in the sandbox vault with the default theme.

SYSTEM INFO:
	Obsidian version: v1.6.7
	Installer version: v1.6.5
	Operating system: Windows 11 Home China 10.0.22631
	Login status: not logged in
	Insider build toggle: off
	Live preview: on
	Base theme: adapt to system
	Community theme: none
	Snippets enabled: 1
	Restricted mode: off
	Plugins installed: 19
	Plugins enabled: 2
		1: BRAT v1.0.1
		2: Note Toolbar v1.10.0-beta-12

RECOMMENDATIONS:
	Custom theme and snippets: for cosmetic issues, please first try updating your theme and disabling your snippets. If still not fixed, please try to make the issue happen in the Sandbox Vault or disable community theme and snippets.
	Community plugins: for bugs, please first try updating all your plugins to latest. If still not fixed, please try to make the issue happen in the Sandbox Vault or disable community plugins.

**Reply by chrisgurney** [reply]
@XMB-7 Thanks again for your help with testing!

If you update to [`beta-13`](https://github.com/chrisgurney/obsidian-note-toolbar/releases/tag/1.10.0-beta-13), I've fixed the issue where you were unable to add new items after adding a separator.

I'm not clear at the moment why the icon to add a separator appears _above_ the _Add toolbar item_ button, and the other strange layout issue you had; I'm unable to reproduce that.
1. Can you also provide an updated recording so I can see where the new line break button appears, as well?
2. _If there's still a problem_ are you at all able to:
  a. inspect via DevTools
  b. select the icon for the Add separator button
  c. take a screenshot or copy/paste the `note-toolbar-setting-items-button-container` DOM element, so I can see what the structure looks like around it?

**Reply by XMB-7** [reply]
After updating to the latest version, everything seems to be working well. Thank you for the quick fixing.

![recording](https://github.com/user-attachments/assets/2bee1572-a0a6-4719-8a39-0df6d5dffb66)

**Reply by chrisgurney** [reply]
@XMB-7 Amazing, thank you! The buttons look like they're in the right place. I'm concerned that your first experience was off, however. No idea what happened there.

**Comment by chrisgurney** [comment]
@XMB-7 I've updated the overall look of settings for breaks/separators, hopefully one last time in [`beta-14`](https://github.com/chrisgurney/obsidian-note-toolbar/releases/tag/1.10.0-beta-14) . I've reversed the look of breaks versus separators, so hopefully it's clearer which is which by glancing at it.

_Would you be able to give this a test to make sure everything's where it is supposed to be? And what do you think?_

<img src="https://github.com/user-attachments/assets/13a36196-e121-4901-9599-8c94aaf8c6bf" alt="Example of a toolbar's settings with line breaks and separators added." width="600"/>

**Reply by XMB-7** [reply]
It works well. The new look is fantastic, and I really think it is clearer and more user-friendly. Thanks for the polish.

**Comment by chrisgurney** [comment]
Closing as separators (and line breaks) are now generally available.
